crates/ra_syntax/test_data/parser/err/0004_use_path_bad_segment.rast

```diff
@@ -9,8 +9,7 @@ SOURCE_FILE@0..12
             NAME_REF@4..7
               IDENT@4..7 "foo"
         COLON2@7..9 "::"
-        PATH_SEGMENT@9..11
-          ERROR@9..11
-            INT_NUMBER@9..11 "92"
+        ERROR@9..11
```

Comment:
Not sure if this is fine...

Reply:
I think it is OK, though the old version was better.  Should be a part of #5006  really, in a sense that we probably should pass-in the parent marker into `err_recover` and let it chose whether to abandon or not. 

bors r+

Thanks!